lib/backend/spotifyService.dart

```diff
@@ -8,98 +8,83 @@ String client_secret = '2d710351a8ef47c18e29c875da325b7f';
 
 Client client = Client();
 
-Future<Response> initialAuth(String code) {
+Future<Response> initialAuth(String code, SharedPreferences prefs) {
   return client.post('https://accounts.spotify.com/api/token', headers:  clientHeaders(), body: preAuthBody(code)).then((Response response){
     final Map<String, dynamic> map = json.decode(response.body);
-    updateAuthToken(map['access_token']);
-    updateRefreshToken(map['refresh_token']);
-    updateExpireTime(map['expires_in'].toString());
-    updateCurrentUser(map['access_token']);
+    updateAuthToken(map['access_token'], prefs);
+    updateRefreshToken(map['refresh_token'], prefs);
+    updateExpireTime(map['expires_in'].toString(), prefs);
+    updateCurrentUser(map['access_token'], prefs);
   });
 }
 
-Future<Response> refreshAuth(String refreshToken) {
-  return client.post('https://accounts.spotify.com/api/token', headers: clientHeaders(), body: refreshBody(refreshToken)).then((Response result) {
-    final Map<String, dynamic> map = json.decode(result.body);
-    updateAuthToken(map['access_token']);
-    updateExpireTime(map['expires_in'].toString());
-    if (map.containsKey('refresh_token')) {
-      updateRefreshToken(map['refresh_token']);
-    }
-  });
+Future<Response> refreshAuth(String refreshToken, SharedPreferences prefs) {
+  return client.post('https://accounts.spotify.com/api/token', headers: clientHeaders(), body: refreshBody(refreshToken))
+      .then((Response result) {
+        final Map<String, dynamic> map = json.decode(result.body);
+        updateAuthToken(map['access_token'], prefs);
+        updateExpireTime(map['expires_in'].toString(), prefs);
+        if (map.containsKey('refresh_token')) {
+          updateRefreshToken(map['refresh_token'], prefs);
+        }
+    });
 }
 
-Future<String> getNowPlaying(String authToken) async {
-  String nowPlaying = 'Unable to get Now Playing';
+Future<Map<String, dynamic>> getNowPlaying(String authToken) async {
```

Comment:
if you want the song id it should just be something like 
`String id = getNowPlaying(auth).then((Map<String, dynamic> map) => map['id'])`
or
`Map<String, dynamic> map = await getNowPlaying(auth); String id = map['id'];`